docs/architecture-overview.md

```diff
@@ -0,0 +1,159 @@
+## Architecture Overview
```

Comment:
NOTE TO REVIEWERS: [the formatted version of this](https://github.com/Baron-Severin/fenix/blob/5799-document-architecture/docs/architecture-overview-outline.md) may be easier to review